### PR TITLE
DOCSP-38286 Enable ORR

### PR DESCRIPTION
**Pull request by ajhuh-mdb**
## DESCRIPTION 
- Creates reference page for ORR 
- Adds note on ORR to 1.8 beta release notes 
- Adds ORR to beta program feature table

## STAGING 
- [Beta Feature table](https://preview-mongodbajhuhmdb.gatsbyjs.io/cluster-sync/DOCSP-38286-enable-ORR/reference/beta-program/#beta-features)
- [ORR reference page](https://preview-mongodbajhuhmdb.gatsbyjs.io/cluster-sync/DOCSP-38286-enable-ORR/reference/beta-program/orr/#std-label-c2c-beta-orr)
- [1.8 RN](https://preview-mongodbajhuhmdb.gatsbyjs.io/cluster-sync/DOCSP-38286-enable-ORR/release-notes/1.8/#oplog-rollover-resilience)

## JIRA 
https://jira.mongodb.org/browse/DOCSP-38286

## BUILD
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=66846c07fe162e5006346204

### CODE REVIEW

File: source/reference/beta-program.txt [hunk 1]
```diff
@@ -73,15 +73,15 @@ Beta Features
...
-     - # DESCRIPTION TBD 
-
    * - :ref:`c2c-beta-destination-data-handling`
      - .. include:: /includes/destinationDataHandling-introduction.rst
 
    * - :ref:`c2c-beta-many-to-one`
      - .. include:: /includes/many-with-one-cluster.rst
 
+   * - :ref:`c2c-beta-orr`
+     - .. include:: /includes/orr-intro.rst
```
Comment (sarah-olson-mongodb) [inline source/reference/beta-program.txt:83]: (question / suggestion)
What is your "organizational strategy" for this table? If featured are just listed in random order, maybe alphabetize, at minimum?

File: source/includes/orr-intro.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Starting in {+c2c-beta-program-short+} 1.8, you can enable Oplog Rollover
```
Comment (sarah-olson-mongodb) [inline source/includes/orr-intro.rst:1]: (consistency)
Here, mongosync is in monospace, while beta is not. Above, in Filter Documents mongosync-beta is in monospace, and has a hyphen. I think you may just need to make Filter Documents consistent with everything else here. 

Preview link for convience: https://preview-mongodbajhuhmdb.gatsbyjs.io/cluster-sync/DOCSP-38286-enable-ORR/reference/beta-program/#beta-features

File: source/includes/orr-intro.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Starting in {+c2c-beta-program-short+} 1.8, you can enable Oplog Rollover 
+Resilience (ORR). With ORR,  ``mongosync`` applies changes made on the source 
+cluster to the destination cluster during the initial sync.
```
Comment (sarah-olson-mongodb) [inline source/includes/orr-intro.rst:3]: (short description)
I wonder if you could rework this last sentence to include a couple of words about why this is useful. You describe it really well in the reference page second paragraph. What do you think about adding some of that to this introduction.

File: source/reference/beta-program/orr.txt [hunk 4]
```diff
@@ -13,3 +13,53 @@ Enable Oplog Rollover Resilience
    :class: singlecol
 
 .. include:: /includes/beta-feature.rst
+
+.. include:: /includes/orr-intro.rst
+
+By applying changes earlier in the sync process, ``mongosync`` maintains a more 
+recent position in the :term:`oplog`. This adds resilience to long-running
```
Comment (sarah-olson-mongodb) [inline source/reference/beta-program/orr.txt:20]: (question)
Is "a more recent position" a jargon-y phrase? Is there a different way to phrase this? I am less familiar with C2C, so I'm genuinely asking. : )

File: source/reference/beta-program/orr.txt [hunk 5]
```diff
@@ -13,3 +13,53 @@ Enable Oplog Rollover Resilience
    :class: singlecol
 
 .. include:: /includes/beta-feature.rst
+
+.. include:: /includes/orr-intro.rst
+
+By applying changes earlier in the sync process, ``mongosync`` maintains a more 
+recent position in the :term:`oplog`. This adds resilience to long-running 
+operations, mitigates the risk of oplog rollover, and significantly lowers 
+the risk of restarting the sync.
```
Comment (sarah-olson-mongodb) [inline source/reference/beta-program/orr.txt:22]: (optional suggestion)
Consider putting these benefits into a bulleted list? Not at all mandatory, but it might make it more scannable.

File: source/reference/beta-program/orr.txt [hunk 6]
```diff
@@ -13,3 +13,53 @@ Enable Oplog Rollover Resilience
...
+
+Syntax 
+------
+
+To enable ORR, use the ``--oplogRolloverResilienceIntervalSeconds`` command line
```
Comment (sarah-olson-mongodb) [inline source/reference/beta-program/orr.txt:27]: (minimalism)
You could omit "the" and "command line option" I think, if you like.

File: source/release-notes/1.8.txt [hunk 7]
```diff
@@ -36,12 +36,19 @@ Document Filtering in mongosync
...
+```````````````````````````````````````````````````````````
 
 .. include:: /includes/destinationDataHandling-introduction.rst
 
 For details, see :ref:`c2c-beta-destination-data-handling`.
 
+Oplog Rollover Resilience 
+`````````````````````````
+
+.. include:: /includes/orr-intro.rst
```
Comment (sarah-olson-mongodb) [inline source/release-notes/1.8.txt:48]: (praise)
Really nice way of handling the intro text of the three locations!

**Review by sarah-olson-mongodb — COMMENTED** [review]
Thanks so much @ajhuh-mdb. I left a couple minor comments for consideration. Let me know when you're ready for a super quick pass.

File: source/reference/beta-program.txt (hunk 1 above)
Comment (ajhuh-mdb) [reply source/reference/beta-program.txt:83]: Hi! This is intended to be alphabetized — the changes made in #332 changes the name of some of these features in the table, so I didn't want to change them here to avoid conflicting changes! For instance, "Synchronize Data Between Multiple Source Clusters Simultaneously" will become "Many-to-One Migrations". Also ABC migrations will be added to the top.

File: source/includes/orr-intro.rst (hunk 2 above)
Comment (ajhuh-mdb) [reply source/includes/orr-intro.rst:1]: Ah gotcha, thanks for catching that. That text in the filter documents page was added before we created a snooty constant for it. Fixed!

File: source/includes/orr-intro.rst (hunk 3 above)
Comment (ajhuh-mdb) [reply source/includes/orr-intro.rst:3]: Noted and updated!

File: source/reference/beta-program/orr.txt (hunk 4 above)
Comment (ajhuh-mdb) [reply source/reference/beta-program/orr.txt:20]: Gotcha — "more recent position" refers to how the `mongosync` operation is recorded higher up in the oplog (hence in a more recent position)! Can't say "most recent operation" since there are likely other change events being logged by the oplog, which is why I ended up with "more recent position."

File: source/reference/beta-program/orr.txt (hunk 5 above)
Comment (ajhuh-mdb) [reply source/reference/beta-program/orr.txt:22]: Opting to keep this one as is. Though the list would make it more scannable, I feel like the need for a list intro sentence might make it more verbose than necessary!

File: source/reference/beta-program.txt (hunk 1 above)
Comment (sarah-olson-mongodb) [reply source/reference/beta-program.txt:83]: Thanks @ajhuh-mdb! This is great!

File: source/includes/orr-intro.rst (hunk 2 above)
Comment (sarah-olson-mongodb) [reply source/includes/orr-intro.rst:1]: Delightful! TY!

File: source/reference/beta-program/orr.txt (hunk 4 above)
Comment (sarah-olson-mongodb) [reply source/reference/beta-program/orr.txt:22]: Ok, got it! Thanks for the explanation. That works for me!

**Review by sarah-olson-mongodb — APPROVED** [review]
Thanks a ton @ajhuh-mdb! 
LGTM

**Comment by ajhuh-mdb** [comment]
Hi @edobranov 👋 Whenever you get the chance, could you review this docs PR for documenting ORR for `mongosync` beta? TIA!

File: source/includes/orr-intro.rst [hunk 8]
```diff
@@ -0,0 +1,6 @@
+Starting in {+c2c-beta-program-short+} 1.8, you can enable Oplog Rollover 
+Resilience (ORR). With ORR,  ``mongosync`` applies changes made on the source 
+cluster to the destination cluster during the initial sync. This adds resilience 
+to long-running operations, mitigates the risk of oplog rollover, and 
+significantly lowers the risk of restarting the sync.
```
Comment (edobranov) [inline source/includes/orr-intro.rst:5]: A couple suggestions on the phrasing for clarity:

For the `With ORR` sentence, the "during the initial sync" part is a bit ambiguous to me and may not fully describe the advantage of ORR to some readers. Even without ORR, mongosync still technically applies the events that occurred _during_ initial sync (without ORR, this application just happens _after_ the initial sync). I think we can make an emphasis that ORR allows mongosync to apply the events _concurrently_ with initial sync:
> With ORR, `mongosync` concurrently applies changes made on the source cluster to the destination cluster as initial sync is ongoing.

For the `This adds` sentence, the "long-running operations" phrase is a bit vague. It's not clear whether it's some user operation on the source, the sync itself, or something else. I'd rephrase it to something like:
> For source clusters with a high write rate, ORR significantly lowers the risk of oplog rollover during initial sync and needing to restart the sync.

File: source/reference/beta-program/orr.txt [hunk 9]
```diff
@@ -13,3 +13,54 @@ Enable Oplog Rollover Resilience
    :class: singlecol
 
 .. include:: /includes/beta-feature.rst
+
+Starting in {+c2c-beta-program-short+} 1.8, you can enable Oplog Rollover 
+Resilience (ORR). With ORR,  ``mongosync`` applies changes made on the source 
+cluster to the destination cluster during the initial sync.
```
Comment (edobranov) [inline source/reference/beta-program/orr.txt:19]: (same remark as above)

File: source/reference/beta-program/orr.txt [hunk 10]
```diff
@@ -13,3 +13,54 @@ Enable Oplog Rollover Resilience
...
+
+By applying changes earlier in the sync process, ``mongosync`` maintains a more 
+recent position in the :term:`oplog`. This adds resilience to long-running 
+operations, mitigates the risk of oplog rollover, and significantly lowers 
+the risk of restarting the sync.
```
Comment (edobranov) [inline source/reference/beta-program/orr.txt:24]: (same remark as above)

File: source/reference/beta-program/orr.txt [hunk 11]
```diff
@@ -13,3 +13,54 @@ Enable Oplog Rollover Resilience
...
+
+Syntax 
+------
+
+To enable ORR, use ``--oplogRolloverResilienceIntervalSeconds``and specify the 
+interval, in seconds, in which ``mongosync`` checks for eligible change events 
+in the oplog. The default value is ``-1``, which disables the option.
```
Comment (edobranov) [inline source/reference/beta-program/orr.txt:31]:
```suggestion
in the oplog. The default value is ``-1``, which disables ORR.
```
An option can't be disabled per se; it's the ORR mechanism that gets disabled.

**Review by edobranov — CHANGES_REQUESTED** [review]
Looks great, just a couple suggestions!

File: source/reference/beta-program/orr.txt (hunk 11 above)
Comment (ajhuh-mdb) [reply source/reference/beta-program/orr.txt:31]: Noted and updating!

**Comment by ajhuh-mdb** [comment]
Thanks for the initial review @edobranov! This should be ready for another look.

File: source/includes/orr-intro.rst [hunk 12]
```diff
@@ -0,0 +1,5 @@
+Starting in {+c2c-beta-program-short+} 1.8, you can enable Oplog Rollover 
+Resilience (ORR). With ORR,  ``mongosync`` applies changes made on the source 
+cluster to the destination cluster concurrently with initial sync. For source 
+clusters with a high write rate, ORR significantly lowers the risk of oplog 
+rollover and reduces the need to restart the sync.
```
Comment (edobranov) [inline source/includes/orr-intro.rst:5]:
```suggestion
rollover during initial sync and reduces the need to restart the sync.
```
Small clarification that ORR is only a benefit during initial sync and not afterwards.

File: source/reference/beta-program/orr.txt [hunk 13]
```diff
@@ -13,3 +13,54 @@ Enable Oplog Rollover Resilience
...
 
 .. include:: /includes/beta-feature.rst
+
+Starting in {+c2c-beta-program-short+} 1.8, you can enable Oplog Rollover 
+Resilience (ORR). With ORR,  ``mongosync`` applies changes made on 
+the source cluster to the destination cluster concurrently with initial sync.
+
+By applying changes earlier in the sync process, ``mongosync`` maintains a more 
+recent position in the :term:`oplog`. For source clusters with a high write 
+rate, ORR significantly lowers the risk of oplog rollover and reduces
```
Comment (edobranov) [inline source/reference/beta-program/orr.txt:23]:
```suggestion
rate, ORR significantly lowers the risk of oplog rollover during initial sync and reduces 
```
(same remark as above)

File: source/reference/beta-program/orr.txt [hunk 14]
```diff
@@ -13,3 +13,54 @@ Enable Oplog Rollover Resilience
...
+
+   ./bin/mongosync \
+         --cluster0 "mongodb://localhost:27000" \
+         --cluster1 "mongodb://localhost:27001" \
+         --oplogRolloverResilienceIntervalSeconds 60
+
+Behavior 
+--------
+
+ORR increases the resilience of ``mongosync`` to oplog rollover but does not
```
Comment (edobranov) [inline source/reference/beta-program/orr.txt:48]:
```suggestion
ORR increases the resilience of ``mongosync`` to oplog rollover during initial sync but does not 
```
(same remark as above)

**Review by edobranov — APPROVED** [review]
LGTM! Just had one final small suggestion, but feel free to ignore / mark as resolved if you think the current text is clear enough. Thanks! 🙌

File: source/includes/orr-intro.rst (hunk 12 above)
Comment (ajhuh-mdb) [reply source/includes/orr-intro.rst:5]: Ah good catch  — updating across all instances!
